File: mcp-server/internal/domains/agents/resources.go
package agents

import (
	"github.com/modelcontextprotocol/go-sdk/mcp"

	"github.com/stigmer/stigmer/mcp-server/internal/domains"
)

// Template returns the MCP resource template for agents.
func Template() *mcp.ResourceTemplate {
	return &mcp.ResourceTemplate{
		URITemplate: "stigmer://agents/{org}/{slug}",
		Name:        "stigmer_agent",
		Title:       "Stigmer Agent",
		Description: "Full definition of a Stigmer agent, identified by organization and slug.",
		MIMEType:    "application/json",
	}
}

// ResourceHandler returns a handler that reads an agent resource by parsing
// the org and slug from the request URI.
func ResourceHandler(serverAddress string) mcp.ResourceHandler {
	return domains.NewResourceHandler(Fetch, serverAddress, "agents")
}

File: mcp-server/src/domains/agents/resources.ts
// Agent resource template (stigmer://agents/{org}/{slug}).
// Go parity: mcp-server/internal/domains/agents/resources.go.

import type { McpServer } from "@modelcontextprotocol/sdk/server/mcp.js";

import type { BackendTarget } from "../client.js";
import { registerResource } from "../resourcehandler.js";
import { fetchAgent } from "./fetch.js";

/** Register the agent resource template; returns the registered resource names. */
export function registerAgentResources(server: McpServer, target: BackendTarget): string[] {
  registerResource(server, target, {
    name: "stigmer_agent",
    title: "Stigmer Agent",
    description: "Full definition of a Stigmer agent, identified by organization and slug.",
    template: "stigmer://agents/{org}/{slug}",
    fetch: fetchAgent,
  });
  return ["stigmer_agent"];
}
